ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/page.tsx

```diff
@@ -1,88 +1,67 @@
 'use client';
 
-import { useState } from 'react';
 import { CodeBracketSquareIcon } from '@heroicons/react/20/solid';
 import { Button } from '@inngest/components/Button';
 
 import { useEnvironment } from '@/app/(dashboard)/env/[environmentSlug]/environment-context';
-import Header from '@/components/Header/Header';
-import LoadingIcon from '@/icons/LoadingIcon';
-import FunctionListNotFound from './FunctionListNotFound';
-import FunctionStateFilter, { type FunctionState } from './FunctionStateFilter';
+import Header, { type HeaderLink } from '@/components/Header/Header';
+import { useBooleanSearchParam } from '@/utils/useSearchParam';
 import { FunctionTable } from './FunctionTable';
-import { useFunctionList } from './useFunctionList';
+import { useRows } from './useRows';
 
 export const runtime = 'nodejs';
 
 export default function FunctionListPage() {
-  // Used to turn non render errors into render errors. This is necessary
-  // because React error boundaries only catch errors that occur during
-  // rendering.
-  const [error, setError] = useState<Error>();
+  const [archived] = useBooleanSearchParam('archived');
+
+  const { error, isLoading, hasMore, loadMore, rows } = useRows({ archived: archived ?? false });
   if (error) {
     throw error;
   }
 
-  const [functionState, setFunctionState] = useState<FunctionState>('active');
-  const environment = useEnvironment();
-
-  const [functionList, setFunctionList] = useFunctionList({
-    environmentID: environment.id,
-    onError: setError,
-    functionState,
-  });
-
-  function loadMore() {
-    setFunctionList((prev) => {
-      return {
-        ...prev,
-        latestRequestedPage: prev.latestRequestedPage + 1,
-      };
-    });
-  }
-
-  const isLoadingInitialData = functionList.latestLoadedPage === 0;
-  const environmentHasFunctions = environment.functionCount > 0;
-
-  let content: JSX.Element;
-  if (isLoadingInitialData) {
-    content = (
-      <div className="flex h-full w-full items-center justify-center">
-        <LoadingIcon />
-      </div>
-    );
-  } else if (environmentHasFunctions === false) {
-    content = <FunctionListNotFound />;
-  } else {
-    content = (
+  return (
+    <>
+      <FunctionsHeader />
       <div className="flex min-h-0 flex-1 flex-col divide-y divide-slate-100">
-        <FunctionStateFilter handleClick={setFunctionState} selectedOption={functionState} />
-        <FunctionTable rows={functionList.rows} />
+        <FunctionTable rows={rows} />
 
-        {functionList.hasNextPage && (
+        {hasMore !== false && (
           <div className="flex w-full justify-center py-2.5">
             <Button
-              disabled={functionList.isLoading}
+              disabled={isLoading}
```

Comment:
```suggestion
              loading={isLoading}
```
Our Buttons have a loading state, which seems better for this situation, as it adds a nice spinner that matches the label you added saying "Loading"